Admin/AbstractAdmin.php

```diff
@@ -2174,10 +2174,13 @@ public function getTranslationDomain()
      */
     public function setTranslator(TranslatorInterface $translator)
     {
-        @trigger_error(
-            'The '.__METHOD__.' method is deprecated since version 3.9 and will be removed in 4.0.',
-            E_USER_DEPRECATED
-        );
+        $args = func_get_args();
+        if (isset($args[1]) && $args[1]) {
```

Comment:
It's validating the second argument. See https://github.com/sonata-project/SonataAdminBundle/pull/4235#discussion_r92720511